core/annotations/annotations.go

```diff
@@ -4,9 +4,73 @@
 package annotations
 
 import (
+	"fmt"
+
 	"github.com/juju/errors"
+	"github.com/juju/names/v5"
+)
+
+// Kind identifies different kinds of entities that'll get associated with annotations.
+type Kind int
+
+const (
+	KindCharm Kind = iota
+	KindMachine
+	KindUnit
+	KindModel
+	KindVolume
+	KindFilesystem
 )
 
+// ID reifies annotatable GlobalEntities into an internal representation using annotations.Kind.
+type ID struct {
+	Kind Kind
+	Name string
+}
+
+// ParseID converts the names.Tag into an ID for different names.Kinds of entities,
+// registering them as annotations.Kinds of entities.
+func ParseID(n names.Tag) (ID, error) {
+	switch n.Kind() {
+	case names.CharmTagKind:
+		return ID{
+			Kind: KindCharm,
+			Name: n.Id(),
+		}, nil
+	case names.MachineTagKind:
+		return ID{
+			Kind: KindMachine,
+			Name: n.Id(),
+		}, nil
+	case names.UnitTagKind:
+		return ID{
+			Kind: KindUnit,
+			Name: n.Id(),
+		}, nil
+	case names.ModelTagKind:
+		return ID{
+			Kind: KindModel,
+			Name: n.Id(),
+		}, nil
+	case names.VolumeTagKind:
+		return ID{
+			Kind: KindVolume,
+			Name: n.Id(),
+		}, nil
+	case names.FilesystemTagKind:
+		return ID{
+			Kind: KindFilesystem,
+			Name: n.Id(),
+		}, nil
+	default:
+		return ID{}, fmt.Errorf("unknown kind %q", n.Kind())
+	}
+}
+
+func (i ID) String() string {
+	return fmt.Sprintf("%v/%v", i.Kind, i.Name)
```

Comment:
It might be better for refactoring purposes to say `"%d/%s"` here that way if the types change we will get  checkers that fail and the meaning doesn't potentially cause a breaking change.

Very nity comment.